packages/app/src/cli/utilities/developer-platform-client/shopify-developers-client.ts

```diff
@@ -0,0 +1,380 @@
+import {PartnersSession} from '../../../cli/services/context/partner-account-info.js'
+import {filterDisabledBetas} from '../../../cli/services/dev/fetch.js'
+import {MinimalOrganizationApp, Organization, OrganizationApp, OrganizationStore} from '../../models/organization.js'
+import {selectOrganizationPrompt} from '../../prompts/dev.js'
+import {ExtensionSpecification} from '../../models/extensions/specification.js'
+import {AllAppExtensionRegistrationsQuerySchema} from '../../api/graphql/all_app_extension_registrations.js'
+import {ActiveAppVersionQuerySchema} from '../../api/graphql/app_active_version.js'
+import {
+  GenerateSignedUploadUrlSchema,
+  GenerateSignedUploadUrlVariables,
+} from '../../api/graphql/generate_signed_upload_url.js'
+import {ExtensionUpdateDraftInput, ExtensionUpdateSchema} from '../../api/graphql/update_draft.js'
+import {AppDeploySchema, AppDeployVariables} from '../../api/graphql/app_deploy.js'
+import {FindStoreByDomainSchema} from '../../api/graphql/find_store_by_domain.js'
+import {AppVersionsQuerySchema} from '../../api/graphql/get_versions_list.js'
+import {ExtensionCreateSchema, ExtensionCreateVariables} from '../../api/graphql/extension_create.js'
+import {
+  ConvertDevToTestStoreSchema,
+  ConvertDevToTestStoreVariables,
+} from '../../api/graphql/convert_dev_to_test_store.js'
+import {FunctionUploadUrlGenerateResponse} from '@shopify/cli-kit/node/api/partners'
+import {randomUUID} from '@shopify/cli-kit/node/crypto'
+import {isUnitTest} from '@shopify/cli-kit/node/context/local'
+import {AbortError, BugError} from '@shopify/cli-kit/node/error'
+import {orgScopedShopifyDevelopersRequest} from '@shopify/cli-kit/node/api/shopify-developers'
+import {getArrayRejectingUndefined} from '@shopify/cli-kit/common/array'
+
+const ORG1 = {
+  id: '1',
+  businessName: 'Test Org',
+}
+
+export class ShopifyDevelopersClient implements DeveloperPlatformClient {
+  private _session: PartnersSession | undefined
+
+  constructor(session?: PartnersSession) {
+    this._session = session
+  }
+
+  async session(): Promise<PartnersSession> {
+    if (!this._session) {
+      if (isUnitTest()) {
+        throw new Error('ShopifyDevelopersClient.session() should not be invoked dynamically in a unit test')
+      }
+      // Need to replace with actual auth
+      this._session = {
+        token: 'token',
+        accountInfo: {
+          type: 'UserAccount',
+          email: 'mail@example.com',
+        },
+      }
+    }
+    return this._session
+  }
+
+  async token(): Promise<string> {
+    return (await this.session()).token
+  }
+
+  async refreshToken(): Promise<string> {
+    return this.token()
+  }
+
+  async accountInfo(): Promise<PartnersSession['accountInfo']> {
+    return (await this.session()).accountInfo
+  }
+
+  async appFromId(_appId: string): Promise<OrganizationApp | undefined> {
+    throw new BugError('Not implemented: appFromId')
```

Comment:
I'm intentionally using `BugError` for these, as we should never run into them in the course of normal usage. We should define these functions before ever calling them; for now we are only even having these functions to conform to the interface.

Also, they're better for development. `AbortError` shows just the error message, while `BugError` reports a full stack trace. So since this client is still in development, `BugError` makes more sense to me.

When we release the client for public use, I agree we shouldn't have `BugError`s in here.

Reply:
Good point. Agreed!